students/BrianB/lesson08/circle.py

```diff
@@ -0,0 +1,147 @@
+#!usr/bin/env python3
+# ---------------------------------------------------------------------
+# The goal is to create a class that represents
+# a simple circle.
+#
+# A Circle can be defined by either specifying
+# the radius or the diameter, and the user can
+# query the circle for either its radius or diameter.
+#
+# Other abilities of a Circle instance:
+#   Compute the circle’s area.
+#   Print the circle and get something nice.
+#   Be able to add two circles together.
+#   Be able to compare two circles to see which is bigger.
+#   Be able to compare to see if they are are equal.
+#   (follows from above) be able to put them in a list and sort them.
+# ---------------------------------------------------------------------
+from math import pi, sqrt
+
+# print("hello world")
+
+
+class Circle(object):
+    def __init__(self, radius):
+        self._radius = radius
+
+    @property
+    def radius(self):
+        return self._radius
+
+    @radius.setter
+    def radius(self, radius):
+        self._radius = radius
+
+    @property
+    def diameter(self):
+        return self._radius*2
+
+    @diameter.setter
+    def diameter(self, value):
+        self._radius = value/2
+
+    @property
+    def area(self):
+        return pi * self._radius**2
+
+    @area.setter
+    def area(self, value):
+        self._radius = sqrt(value / pi)
+
+    @classmethod
+    def from_diameter(circle_object, value):
```

Comment:
Since this is a class method, you may want to rename "circle_object" because this could technically be a sphere too. Works 100% functionally but just a consideration for reading code clarity